application_sandbox/extended_dynamic_state2/main.cpp

```diff
@@ -0,0 +1,412 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "application_sandbox/sample_application_framework/sample_application.h"
+#include "support/entry/entry.h"
+#include "vulkan_helpers/buffer_frame_data.h"
+#include "vulkan_helpers/helper_functions.h"
+#include "vulkan_helpers/vulkan_application.h"
+#include "vulkan_helpers/vulkan_model.h"
+
+#include <chrono>
+#include "mathfu/matrix.h"
+#include "mathfu/vector.h"
+
+using Mat44 = mathfu::Matrix<float, 4, 4>;
+using Vector4 = mathfu::Vector<float, 4>;
+
+namespace cube_model {
+#include "cube.obj.h"
+}
+
+// red cube
+const struct {
+  size_t num_vertices;
+  float positions[24];
+  float uv[16];
+  float normals[24];
+  size_t num_indices;
+  uint32_t indices[36];
+} cube_data = {8,
```

Comment:
I would call this red_cube_data, and green_cube_data below (instead of cube_data2).